### PR TITLE
TIC TAC TOE game (project on JS part of the curriculum)

### PR DESCRIPTION
**Pull request by Abdoulaye-Thespy**
In this milestone, I created a game, (TIC TAC TOE) Using vanilla JavaScript, HTML CSS and bootstrap.
Players will need to enter their names and the system will prompt their respective signs for the game and the last screen will be where the game will actually be played.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,32 @@
...
+## Built with
+
+- HTML5
+- CSS3
+- Boostrap
+
+## Getting Started
+
+- Clone the git repository.
+- cd into the project
```
Comment (uryelah) [inline README.md:23]: - [ ] Then open which file?

File: dom.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    const topMiddle = cellBloc[1].classList[3];
+    const topRight = cellBloc[2].classList[3];
+    const middleLeft = cellBloc[3].classList[3];
+    const middleMiddle = cellBloc[4].classList[3];
+    const middleRight = cellBloc[5].classList[3];
+    const botomleft = cellBloc[6].classList[3];
+    const botomMiddle = cellBloc[7].classList[3];
+    const botomRight = cellBloc[8].classList[3];
+
+    if ((topLEft && topLEft === topMiddle && topLEft === topRight) || (topLEft && topLEft === middleLeft && topLEft === botomleft) || (topLEft && topLEft === middleMiddle && topLEft === botomRight)) {
```
Comment (uryelah) [inline dom.js]: - [ ] [OPTIONAL] The logic is working but can you think in a way to achieve the same result, in the win cheking, while repeating less code?

File: dom.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+   
+  </div>
+  `;
+  };
+
+
+  const chooseSign = () => {
+    const playerOne = document.getElementById('uname').value;
+    const playerTwo = document.getElementById('uname2').value;
+    window.play1 = playerOne;
```
Comment (uryelah) [inline dom.js]: - [ ] Please don't use global variables. When you assign a value to a window attribute you are using a global variable in the window scope, which is the most global one in a browser. 
You should instead be creating a Player with a name attribute, that can come from a PlayerFactory, and then you can store the players in the Game object.

File: dom.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        checkWin();
+        next();
+      }
+    }
+  };
+
+
+  const gameBoard = () => {
+    isX = true;
+    cont.innerHTML = `
```
Comment (uryelah) [inline dom.js]: - [ ] Don't mix game logic with DOM manipulation.

File: dom.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+  `;
+    } else {
+      welcome();
+    }
+  };
+
+
+  const change = (e) => {
+    const allClass = e.target.classList;
```
Comment (uryelah) [inline dom.js]: - [ ] Don't mix game logic with DOM manipulation.

File: dom.js (hunk 3 above)
Comment (uryelah) [reply dom.js]: - [ ] Don't mix game logic with DOM manipulation too!

File: dom.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+   <div class="cell cellPlay w"> w</div>
+   <div class="cell cellPlay w">w </div>
+   <div class="cell cellPlay w">w </div>
+   <div class="cell cellPlay w"> w</div>
+  </div>
+   
+  </div>
+  `;
+
+    if (sign === 'x') {
```
Comment (uryelah) [inline dom.js]: - [ ] Don't mix game logic with DOM manipulation.

File: dom.js [hunk 7]
```diff
@@ -0,0 +1,222 @@
+
+const display = (() => {
+  const cont = document.getElementById('game');
+  const cellBloc = document.getElementsByClassName('cell');
+  let isX = true;
+  let sign = null;
+
+  const next = () => {
+    if (isX) {
+      document.getElementById('play').innerHTML = play1;
```
Comment (uryelah) [inline dom.js]: - [ ] Don't mix game logic with DOM manipulation.

File: dom.js [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+    if (sign === 'x') {
+      document.getElementById('winner').innerHTML = `${play1} WON!!!!`;
+    } else if (sign === 't') {
+      document.getElementById('winner').innerHTML = `${play2} WON!!!!`;
+    }
+  };
+
+
+  const checkWin = () => {
+    const topLEft = cellBloc[0].classList[3];
```
Comment (uryelah) [inline dom.js]: - [ ] The Gameboard cells should be stored in an array, please don't just have the logic with the DOM.
render_requirement

File: dom.js (hunk 3 above)
Comment (uryelah) [reply dom.js]: objects_requirement

**Review by uryelah — CHANGES_REQUESTED** [review]
Hi **Abdoulaye Razak**, I'm Sarah your reviewer for this PR 👩‍🚀 

Good job with making the game work and having a live version. 
Sadly there are big changes you'll need to implement in order to pass the requirements of this project.

You have the DOM manipulation, when you add/remove/change elements in the page, mixed with the game logic while the game logic itself is missing the Player/Game/GameBoard objects.

This project follows the lesson about Factory Functions and The Module Pattern so the main goal is to put into practice those concepts + OOP and accessing object attributes.

[I recommend that you re-read the Pathwright instructions carefully.](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623090/)

**The main takeaway is to separate logic from DOM**, but make sure to check the other required fixes too.

Rock on! 🎸 👨‍🎤 

# Required Changes 🏗️ 🐛 💡

- [ ] Make sure all the linter checks are passing.

- [ ] Please de-couple the game logic from the DOM manipulation. 
You should in theory be able to play the full game in the console, no DOM, just with the logic made. 
The UI, or DOM, is just the top layer to make the logic interact and be able to get input from the user directly.
You already have two files, you can separate the logic in one of them or, if you decide to go with one single file, separate the logic at the top of the file and the DOM manipulation at the bottom.

- [ ] Use factory functions or classes, constructors, to instantiate the Player, the Game, and the Board objects from.

- [ ] Make sure the Player names are stored at the Player objects, not just inside DOM elements.

- [ ] Store the player moves in an array inside of a Gameboard.

- [ ] Have a function responsible only for rendering the contents of the gameboard array to the webpage

- [ ] Make sure the top buttons you added work, on the local and deploy version, or remove them otherwise.

(Check the comments below for the other fixes)

-----

 **Kindly fix the pointed out issues and comments below before requesting the next code review**

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

File: dom.js (hunk 4 above)
Comment (wathigo) [reply dom.js]: Not done!
`gameBoard` should an object. We can use a define it in a separate file and then import it to improve code readability.
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/objects_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript)).
It uses an array inside of a Gameboard object to store the 'moves' ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/objects_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))

**Review by wathigo — CHANGES_REQUESTED** [review]
### Changes Required
Almost all the previously suggested changes have not been met.
I have attached some screenshots for these requirements.
Feel free to reach out to any of the Full-Time TSEs for assistance.

**REQUIRED**
* Game logic should be separated from DOM manipulative code. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright).
* It has a function with the only responsibility of rendering the contents of the gameboard array to the webpage ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/render_requirement.png) from Odin).
* The game does not take care of a draw.
Screenshot from 2020-09-14 10-37-13.

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo.**

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Helllo @Abdoulaye-Thespy 
### You have done well so far :clap: However, you still have some few issues to fix:-
- [ ] Take care of the game in case it is a draw
- [ ] Solve the linting errors in the pull request
- [ ] Remove the commented lines of code in the app
### Make the changes and submit for another code review. Good luck.
- Happy coding!

File: index.html [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+   <div class="cell"> </div>
+   <div class="cell"> </div>
+   <div class="cell"> </div>
+   <div class="cell"> </div>
+   <div class="cell"> </div>
+   <div class="cell"> </div>
+  </div>
+   
+  </div> -->
+
```
Comment (Dandush03) [inline index.html]: - [ ] __Remove Comented Code__

**Review by Dandush03 — CHANGES_REQUESTED** [review]
# Status: :red_circle: Require Changes :red_circle: 
### Great work you almos done! fix this small issue before getting approved.
- [ ] __Remove Commented Code:__ you're allowed to have comments on the code explaining or dividing each section. but if you have a comment code then you should delete it and if needed consult your git log

>I would have approved this project if this issue wasn't mention on each of the previews reviews, and marked as optional so you can continue with your project

# Happy Coding :nerd_face: :computer:

**Review by addod19 — APPROVED** [review]
## Status: Approved :trophy: :balloon: 

Hi @Abdoulaye-Thespy, congratulations on fixing all the comments outlined by the previous TSE's :100: Great job, I hereby approve the project :fireworks: 

[Optional]
- It would be nice to have a separate file for `gameboard` and `player`

Good luck with your next projects :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Pull request merged**